engine/table/src/main/java/io/deephaven/engine/util/TableTools.java

```diff
@@ -768,6 +769,39 @@ public static Table newTable(TableDefinition definition, ColumnHolder<?>... colu
         };
     }
 
+    /**
+     * Creates a metadata Table, as represented by {@code definition}. Consists of a string "Name" column from
+     * {@link ColumnDefinition#getName()}, a string "DataType" column from {@link ColumnDefinition#getDataType()}
+     * ({@link Class#getCanonicalName()} if not null, otherwise {@link Class#getName()}), a string "ColumnType" column
+     * from {@link ColumnDefinition#getColumnType()}, and a Boolean "IsPartitioning" column from
+     * {@link ColumnDefinition#isPartitioning()}.
+     *
```

Comment:
Maybe we should add `ComponentType` at this time?

Reply:
I think we can follow up if we think it's important.